src/main/java/org/broadinstitute/hellbender/utils/hmm/interfaces/AlleleMetadataProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.broadinstitute.hellbender.utils.hmm.interfaces;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.vcf.VCFHeader;
+
+import javax.annotation.Nonnull;
+
+/**
+ * An interface for hidden states which are construable as {@link Allele}.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public interface AlleleMetadataProvider {
```

Comment:
I renamed all three "provider" interfaces to "producer". These interfaces can be implemented by certain hidden states. The idea is that the HMM post-processor will require these three interfaces to be implemented. I will think about bundling these interfaces, perhaps into one, soon.